### ISSUE TITLE
document what different files of mutt source code do

### ISSUE THREAD

**Issue by ghost** [neomutt/neomutt.github.io#26]
I think it makse sense to write up what some files in the mutt source code do at the moment, because its not exactly clear from the name what they do. Of course you can add everything you think deserves a place on this list.

*  `crypt-mod.c`vs. `crypt.c` vs. `cryptglue.c`
*  `handler.c`
* `lib.c`
* `main.c` vs `init.c`
* `mbyte.c`
* `muttlib.c`
* `mx.c`
* `recvcmd.c`
* `resize.c`
* `signal.c`
* `strcasestr.c`
* `wcscasecmp.c`
* `wcwidth.c`
* `sendlib.c`
* `base64.c`

@neomutt/reviewers Although this is not a reviewing topic i think you understand the code best of the neomutt team.

The purpose of this issue is to collect knowledge about the code - integrating that into the website is a different issue.

**Comment by ghost** [comment]
Would it make sense also to describe the meaning of un-expressive variables/parameters which at the moment we have not the time to refactor? I honestly don't know the answer, because that could go likewise as a comment into the code.

An Example for that would be for instance: [L1276 of pager.c](https://github.com/neomutt/neomutt/blob/master/pager.c#L1276)

**Comment by gahr** [comment]
Well of course it you're reviewing a change on that line, out of context, and with little clue on what that code does, yeah, k1 bears little semantics by itself. But if you spend 10 minutes to look at the surrounding 50 lines of code, it's pretty clear what this various k* variables do there..
This is to say, yep to improve readability, but let's not overdo..

**Comment by guyzmo** [comment]
@toogley I'm totally with @gahr on this. Don't try to overdocument.

The `k` like one letter variables deserves a better naming, there's no doubt about that, but its usage can be understood from the surrounding code.

About the issue, documenting files should be done as part of the doxygen documentation task. And should happen along with the refactoring. Basically files like `lib.c` and `muttlib.c` are just sinks that get everything that don't belong elsewhere.

All the `strcasestr.c`, `wcscasecmp.c` etc. are just compat sources that get conditionally compiled when the system you're building on does not provide an alternative. I'd be very happy to see them moved to a subfolder called `compat` where `autoconf` would look up for them so it's out of our sight.

Basically, this is a non-issue and should get closed. A simple doxygen header in all the source files we care about will be enough as documentation, and will be generated when we'll start building the doxygen stuff.

**Comment by guyzmo** [comment]
tl;dr I vote to close this issue.,

**Comment by ghost** [comment]
@gahr Right, i haven't looked at the code for 10 minutes (only about 2 or so) - but i sometimes find it hard to understand such code (i don't have much experience with C, rather with Java and such). The point is: i thought this would also help newbies to faster understand mutt's source code.

@guyzmo ah, i didn't knew doxygen can also handle that very well. So yeah, i agree doxygen is a better idea.
